nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/io/LogEntrySerializationTest.java

```diff
@@ -0,0 +1,145 @@
+
+        entry.setExtendedInfos(extendedInfo);
+
+        ObjectMapper mapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addSerializer(ExtendedInfo.class, new ExtendedInfoSerializer());
+        mapper.registerModule(module);
+
+        String json = mapper.writeValueAsString(entry);
+        assertNotNull(json);
+
+        JsonNode jsonLogEntry = mapper.readTree(json);
+        assertEquals(1L, jsonLogEntry.get("id").asLong());
+
+        JsonNode jsonExtendedInfo = jsonLogEntry.get("extended");
+
+        assertEquals(2L, jsonExtendedInfo.get(LONG_INFO).asLong());
+        assertEquals(2.0, jsonExtendedInfo.get(DOUBLE_INFO).asDouble(), 0);
+        assertEquals("this is an info", jsonExtendedInfo.get(STRING_INFO).asText());
+        assertEquals(true, jsonExtendedInfo.get(BOOL_INFO).asBoolean());
+        assertEquals("2012-12-21T00:00:00Z", jsonExtendedInfo.get(DATE_INFO).asText());
+
+        Serializable obj = ExtendedInfoDeserializer.deserializeFromByteArray(jsonExtendedInfo.get(BLOB_INFO).binaryValue());
+        assertEquals("I'm a blob!", ((StringBlob) obj).getString());
+
+    }
+
+    @Test
+    public void testLogEntryDeserialization() throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(ExtendedInfo.class, new ExtendedInfoDeserializer());
+        mapper.registerModule(module);
+
+        ObjectNode logEntryJson = mapper.createObjectNode();
+
+        logEntryJson.put("id", 1L);
+        logEntryJson.with("extended").put(LONG_INFO, 2L);
+        logEntryJson.with("extended").put(DOUBLE_INFO, 42.0);
+        logEntryJson.with("extended").put(BOOL_INFO, false);
+        logEntryJson.with("extended").put(STRING_INFO, "myInfo");
+        logEntryJson.with("extended").put(DATE_INFO, "2012-01-01T00:00:00Z");
+        logEntryJson.with("extended").put("blobInfo",
+                ExtendedInfoSerializer.serializeToByteArray(new StringBlob("I'm a blob!")));
+
+        LogEntryImpl logEntry = mapper.convertValue(logEntryJson, LogEntryImpl.class);
+
+        Map<String, ExtendedInfo> infos = logEntry.getExtendedInfos();
+        assertTrue(infos.get(LONG_INFO) instanceof ExtendedInfoImpl.LongInfo);
+        assertEquals(2L, infos.get(LONG_INFO).getSerializableValue());
+
+        assertTrue(infos.get(DOUBLE_INFO) instanceof ExtendedInfoImpl.DoubleInfo);
+        assertEquals(42.0, infos.get(DOUBLE_INFO).getSerializableValue());
+
+        assertTrue(infos.get(BOOL_INFO) instanceof ExtendedInfoImpl.BooleanInfo);
+        assertEquals(false, infos.get(BOOL_INFO).getSerializableValue());
+
+        assertTrue(infos.get(STRING_INFO) instanceof ExtendedInfoImpl.StringInfo);
+        assertEquals("myInfo", infos.get(STRING_INFO).getSerializableValue());
+
+        assertTrue(infos.get(DATE_INFO) instanceof ExtendedInfoImpl.DateInfo);
+
```

Comment:
We need an actual value check for the date here as well, for instance checking the date timestamp value (milliseconds since epoch).